core/src/main/java/amino/run/oms/migrationDecision/MetricWatcher.java

```diff
@@ -0,0 +1,287 @@
+package amino.run.oms.migrationDecision;
+
+import amino.run.common.MicroServiceID;
+import amino.run.common.MicroServiceNotFoundException;
+import amino.run.common.MicroServiceReplicaNotFoundException;
+import amino.run.common.ReplicaID;
+import amino.run.kernel.common.KernelOID;
+import amino.run.kernel.common.KernelServerNotFoundException;
+import amino.run.kernel.common.metric.NodeMetric;
+import amino.run.kernel.common.metric.RPCMetric;
+import amino.run.oms.KernelServerManager;
+import amino.run.oms.MicroServiceManager;
+import amino.run.policy.DefaultPolicy;
+import amino.run.policy.MigrationNotification;
+import amino.run.policy.util.ResettableTimer;
+import amino.run.runtime.EventHandler;
+import java.net.InetSocketAddress;
+import java.rmi.RemoteException;
+import java.util.*;
+import java.util.logging.Logger;
+
+/**
+ * Class watch over metric collected in {@link MicroServiceManager} and {@link KernelServerManager}.
+ * For evaluating best candidate for migration it maintains a heap based on time consumption in
+ * Micro service replica.
+ *
+ * @author AmitRoushan
+ */
+public class MetricWatcher {
+    private Logger logger = Logger.getLogger(MetricWatcher.class.getName());
```

Comment:
Don't need logger for eacg instance of it. Make it private static final.

Reply:
Addressed in #793 PR